tools/statplot.py

```diff
@@ -148,46 +155,60 @@ def FormatAxis(self, ax2fmt, scale):
                                             color='xkcd:black')
         self.fig.set_tight_layout(True)
 
+    def InitCombo(self, action, prevIterX=None, prevIterY=None):
+        if action == 'load':
+            if 'ElapsedTime' in self.entries:
+                iterX = self.xCombo.get_model().get_iter(
+                    self.entries.index('ElapsedTime'))
+                self.xCombo.set_active_iter(iterX)
+                self.yCombo.set_active(0)
+            else:
+                self.xCombo.set_active(0)
+                self.yCombo.set_active(1)
+        elif action == 'reload':
+            self.xCombo.set_active_iter(prevIterX)
+            self.yCombo.set_active_iter(prevIterY)
+
     def InitCompletion(self, comboBox):
         completion = Gtk.EntryCompletion.new()
         completion.set_text_column(0)
         completion.set_inline_completion(True)
-        completion.set_inline_selection(True)
+        completion.set_inline_selection(False)
         completion.set_model(comboBox.get_model())
         comboBox.get_child().set_completion(completion)
 
     def KeyPressed(self, widget, event):
         key = event.string
         if key == 'r':
-            self.RefreshData(self.statfile)
+            self.RefreshData()
             self.PlotData('update', self.PlotType, self.ax.get_xscale(),
                           self.ax.get_yscale())
         elif key == 'q':
             self.destroy()
         elif key == 'x' or key == 'y':
             if key == 'x':
                 self.get_scale = self.ax.get_xscale
-                self.Data = self.xData
+                curData = self.xData
             elif key == 'y':
                 self.get_scale = self.ax.get_yscale
-                self.Data = self.yData
-            if self.get_scale() == 'linear' \
-                    and self.Data[self.Data != 0].size == 0:
+                curData = self.yData
+            if (self.get_scale() == 'linear'
+                    and curData[curData != 0].size == 0):
```

Comment:
I'd suggest changing this to `(curData == 0).any()` just to make the intention clearer (took me a while to figure out what this was actually doing!)

Reply:
Thanks for that, and sorry for the headache! I have used all instead of any because matplotlib can automatically handle cases with only a few zeros.